src/System.Linq/src/System/Linq/Contains.cs

```diff
@@ -12,16 +11,34 @@ public static partial class Enumerable
         public static bool Contains<TSource>(this IEnumerable<TSource> source, TSource value)
         {
             ICollection<TSource> collection = source as ICollection<TSource>;
-            if (collection != null) return collection.Contains(value);
+            if (collection != null)
+            {
+                return collection.Contains(value);
```

Comment:
Not relevant to this PR, but I was led to think about it: Do we consider the fact that this isn't quite as one might expect from if `collection` is a `HashSet<T>` with a custom comparer, a bug?